src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -490,5 +347,31 @@ private void Dispose(bool disposing)
                 fsWatcher.Dispose();
             }
         }
+
+        public class Startup
+        {
+            public Startup(IConfiguration configuration)
+            {
+                Configuration = configuration;
+            }
+
+            public IConfiguration Configuration { get; }
+
+            // This method gets called by the runtime. Use this method to add services to the container.
+            public IServiceProvider ConfigureServices(IServiceCollection services)
+            {
+                services.AddWebJobsScriptHost();
+
+                services.AddSingleton<ILoggerFactoryBuilder, ConsoleLoggerFactoryBuilder>();
```

Comment:
This is how the CLI logger is injected. This solves the issues with had with logger injection.